src/sentry/models/projectteam.py

```diff
@@ -21,9 +21,11 @@ def get_for_teams_with_org_cache(self, teams: Sequence["Team"]) -> QuerySet["Pro
             .order_by("project__name", "project__slug")
             .select_related("project")
         )
-
         # TODO(dcramer): we should query in bulk for ones we're missing here
-        orgs = {i.organization_id: i.organization for i in teams}
+        orgs = {}
+        for team in teams:
+            if team.organization_id not in orgs:
+                orgs[team.organization_id] = team.organization
```

Comment:
Since we're here, let's make this a bulk query to fetch all the orgs at once?

https://github.com/getsentry/sentry/blob/9c9c320255d27c633b117767eae28aa359519035/src/sentry/api/serializers/models/group.py#L202

Then there's no need for the orgs dict at all. Just iterate through all the project teams and do `set_cached_field_value` with the org directly

Reply:
there will only ever be one org as where this is called asserts that, but i'll make this change

Reply:
Yeah, it's simpler to just stick the prefetch in so that if someone uses this elsewhere it works